backend/src/agent/persistence/client/token_refresher.go

```diff
@@ -0,0 +1,77 @@
+package client
+
+import (
+	log "github.com/sirupsen/logrus"
+	"os"
+	"sync"
+	"time"
+)
+
+type TokenRefresherInterface interface {
+	GetToken() string
+	RefreshToken()
+}
+
+const SA_TOKEN_FILE = "/var/run/secrets/kubeflow/tokens/persistenceagent-sa-token"
```

Comment:
There's a single instance of persistence agent. How would this token file work? Won't multi users share the same token file?

Reply:
Thank you for the review! 

Currently, every pod has an occasioned Service Account and SA token dynamically loaded at `/var/run/secrets/kubernetes.io/serviceaccount/token`.This token has a target audience set to - used to authenticate against the Kube API server -  "https://kubernetes.default.svc.cluster.local"
```
{
  "aud": [
    "https://kubernetes.default.svc.cluster.local"
  ],
  "exp": 1720267473,
  "iat": 1688731473,
  "iss": "https://kubernetes.default.svc.cluster.local",
  "kubernetes.io": {
    "namespace": "kubeflow",
    "pod": {
      "name": "ml-pipeline-persistenceagent-5f9978c49d-npqkn",
      "uid": "2b132bfc-bd86-460d-95ca-00b8c55d7949"
    },
    "serviceaccount": {
      "name": "ml-pipeline-persistenceagent",
      "uid": "d64595e1-53a6-4ada-be90-10398316ba38"
    },
    "warnafter": 1688735080
  },
  "nbf": 1688731473,
  "sub": "system:serviceaccount:kubeflow:ml-pipeline-persistenceagent"
}
```
The new SA Token introduced in this PR and made available to persistenceagent pod via the Service Account Token Volume Projection is time-limited, audience-bound token. Both tokens has the same identity (peristenceagent-pod) but have different audience group. Kubelet takes care to rotate the new token. If the pod recreated , the new token will be recreated too. This makes the need of having "refresh token " logic.  If we decrypt it it looks like this:  
```
{
  "aud": [
    "pipelines.kubeflow.org"
  ],
  "exp": 1688734989,
  "iat": 1688731389,
  "iss": "https://kubernetes.default.svc.cluster.local",
  "kubernetes.io": {
    "namespace": "kubeflow",
    "pod": {
      "name": "ml-pipeline-persistenceagent-5f9978c49d-npqkn",
      "uid": "2b132bfc-bd86-460d-95ca-00b8c55d7949"
    },
    "serviceaccount": {
      "name": "ml-pipeline-persistenceagent",
      "uid": "d64595e1-53a6-4ada-be90-10398316ba38"
    }
  },
  "nbf": 1688731389,
  "sub": "system:serviceaccount:kubeflow:ml-pipeline-persistenceagent"
}
  ```
  Your question a valid. If only persistence-agent should be able to call PipelineAPI's report (scheduled) workflows endpoints and it is just a single service..   is there a need to care about MULTIUSER mode here?

Reply:
Thank you for such detailed reply. 

> With my current implementation, only the Persistence-agent will be able to call these two KF pipeline endpoints. No one else has permission to do that.

> only with these two KFP endpoints there is no need to care about the MULTIUSER mode value - instead, we always should validate that persistent-agent makes the requests. The real user and the namespace are validated when the run/jobs are created.

This sounds good to me!

Reply:
Other than this, is there a way we can add some unit tests to the change? Otherwise, we could accidentally break it without any alert.

Reply:
Unit tests added